command/meta.go

```diff
@@ -434,9 +434,28 @@ func (m *Meta) outputShadowError(err error, output bool) bool {
 	return true
 }
 
+// EnvironmentNameEnvVar is the name of the environment variable (ie, the POSIX
+// feature) that can be used to set the name of the Terraform environment
+// (overriding the environment chosen by `terraform env select`).  Note that
+// this environment variable is ignored by `terraform env new` and `terraform
+// env delete`.
+const EnvironmentNameEnvVar = "TF_ENVIRONMENT"
+
 // Env returns the name of the currently configured environment, corresponding
 // to the desired named state.
 func (m *Meta) Env() string {
+	current, _ := m.EnvWithIsOverridden()
+	return current
+}
+
+// Env returns the name of the currently configured environment, corresponding
+// to the desired named state, as well as a bool saying whether this was
+// set via the TF_ENVIRONMENT environment variable.
+func (m *Meta) EnvWithIsOverridden() (string, bool) {
```

Comment:
It probably doesn't matter, and naming is hard, but I did read this function name with the wrong idea of what it was too many times. Maybe just `EnvOverridden`?